service/src/main/java/bio/terra/externalcreds/services/ProviderService.java

```diff
@@ -122,6 +128,52 @@ public LinkedAccountWithPassportAndVisas createLink(
     return linkedAccountService.saveLinkedAccount(extractPassport(userInfo, linkedAccount));
   }
 
+  public void deleteLink(String userId, String providerId) {
```

Comment:
Should this be a transaction so that we ensure nothing changes with the DB record between our calls to  `getLinkedAccount` and `deleteLinkedAccount`? I know that would leave it hanging for a bit though while we revoke the access token.

Reply:
We should avoid making external api calls within a transaction.  The worst that can happen here is that a user concurrently deletes and creates their link. If the create happens after the delete reads the refresh token but before it deletes the link then the new refresh token obtained by create process will be lost. Not a big deal.

I also want to make sure that it is known that in general transactions don't give this kind of protection anyway. The serializable isolation level is the thing that would give protection.